common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/LwM2mTransportUtil.java

```diff
@@ -323,6 +341,65 @@ public static SimpleDownlinkRequest createWriteAttributeRequest(String target, O
         }
     }
 
+
+    public static void validateVersionedId(LwM2mClient client, HasVersionedId request) {
+        Set<String> msgException = new HashSet<>();
```

Comment:
This method is completely wrong and inefficient. Please rewrite it without creating a HashSet each time you validate something.